portal-api/src/cron-scheduler-jobs/helpers/utilisation-report-helpers.ts

```diff
@@ -0,0 +1,203 @@
+      console.warn(`Not sending ${emailDescription} email to '${bankName}' - paymentOfficerTeam.emails property against bank is not an array or is empty`);
+      return;
+    }
+    await Promise.all(
+      paymentOfficerTeamEmails.map(async (paymentOfficerTeamEmail) => {
+        if (!isValidEmail(paymentOfficerTeamEmail)) {
+          console.error(`Failed to send ${emailDescription} email to '${paymentOfficerTeamEmail}' - invalid payment officer email`);
+        } else {
+          await sendEmailCallback(paymentOfficerTeamEmail, getEmailRecipient(paymentOfficerTeam, bankName), formattedReportPeriod);
+          console.info(`Successfully sent '${emailDescription}' email to '${bankName}' (bank ID: ${bankId})`);
+        }
+      }),
+    );
+  } catch (error) {
+    console.error(`Failed to send ${emailDescription} email for bank '${bankName}':`, error);
+  }
+};
+
+/**
+ * For a given bank, check if a utilisation report has been received for the current reporting period
+ * then, where not yet received, attempts to call the provided callback function to send the required email
+ * @param emailDescription {string} - a description of the email (used for logging)
+ * @param sendEmailCallback {SendEmailCallback} - callback function that sends the required email.
+ *   The callback is provided the following argument `{ emailAddress: string; recipient: string }`,
+ *   extracted from the provided bank.
+ * @returns {Promise<void>}
+ */
+const sendEmailToBankIfReportNotReceived = async (bank: BankResponse, emailDescription: string, sendEmailCallback: SendEmailCallback): Promise<void> => {
+  const { name: bankName, id: bankId, utilisationReportPeriodSchedule: schedule } = bank;
+  const currentReportingPeriodForBank = getCurrentReportPeriodForBankSchedule(schedule);
+  const formattedReportPeriod = getFormattedReportPeriod(currentReportingPeriodForBank);
+
+  if (isPreviousCalendarMonth(currentReportingPeriodForBank.end.month)) {
+    const isReportDue = await getIsReportDue(bankId, currentReportingPeriodForBank);
+
+    if (isReportDue) {
+      await sendEmailForBank(emailDescription, formattedReportPeriod, sendEmailCallback, bank);
+    } else {
+      console.info(
+        `Not sending '${emailDescription}' email to '${bankName}' (bank ID: ${bankId}) - report has already been submitted for ${formattedReportPeriod} report period`,
+      );
+    }
+  } else {
+    console.info(
+      `Not sending '${emailDescription}' email to '${bankName}' (bank ID: ${bankId}) - report is not due this month, current reporting period for bank is ${formattedReportPeriod}`,
+    );
+  }
+};
+
+/**
+ * For all banks, checks if the current utilisation report period for bank is in it's submission period and
+ * if so if utilisation report has been received for the current reporting period
+ * then, where not yet received, attempts to call the provided callback function to send the required email
```

Comment:
I think this description is a little tricky to read. Is there any way we could re-phrase it to make it a bit clearer?